mev-boost-relay/services/api/proofs.go

```diff
@@ -20,37 +19,23 @@ var (
 )
 
 // verifyInclusionProof verifies the proofs against the constraints, and returns an error if the proofs are invalid.
-func verifyInclusionProof(log *logrus.Entry, transactionsRoot phase0.Root, proof *common.InclusionProof, constraints map[phase0.Hash32]*Constraint) error {
+func verifyInclusionProof(log *logrus.Entry, transactionsRoot phase0.Root, proof *common.InclusionProof, hashToConstraints HashToConstraintDecoded) error {
 	if proof == nil {
 		return ErrNilProof
 	}
 
+	constraints := ParseConstraintsDecoded(hashToConstraints)
+
 	leaves := make([][]byte, len(constraints))
 
-	i := 0
-	for hash, constraint := range constraints {
+	for i, constraint := range constraints {
 		if constraint == nil {
 			return ErrNilConstraint
 		}
 
-		if len(constraint.Tx) == 0 {
-			log.Warnf("[BOLT]: Raw tx is empty for constraint tx hash %s", hash)
-			continue
-		}
-
 		// Compute the hash tree root for the raw preconfirmed transaction
 		// and use it as "Leaf" in the proof to be verified against
-
-		// TODO: this is pretty inefficient, we should work with the transaction already
-		// parsed without the blob here to avoid unmarshalling and marshalling again
-		transaction := new(types.Transaction)
-		err := transaction.UnmarshalBinary(constraint.Tx)
-		if err != nil {
-			log.WithError(err).Error("error unmarshalling transaction while verifying proofs")
-			return err
-		}
-
-		withoutBlob, err := transaction.WithoutBlobTxSidecar().MarshalBinary()
+		withoutBlob, err := constraint.Tx.MarshalBinary()
```

Comment:
If this is without blob, how do you get the encoding with blob?

Reply:
Good catch, actually when calling this functions the constraints txs are already forwarded without blobs (see https://github.com/chainbound/bolt/pull/162/files/880cc52196201062f1021ee9db549b93109ed00a#diff-f03b62c88edc3b36b56c7ee1b08c7f6670b22e21c52d18430f9f1d7f3eff7534), however the variable name has been left the same which is a bit misleading